sagemaker-clarify/fairness_and_explainability/fairness_and_explainability_boto3.ipynb

```diff
@@ -0,0 +1,1541 @@
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "training_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, let's upload the data to S3."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.s3 import S3Uploader\n",
+    "from sagemaker.inputs import TrainingInput\n",
+    "\n",
+    "train_uri = S3Uploader.upload(\"train_data.csv\", \"s3://{}/{}\".format(bucket, prefix))\n",
+    "train_input = TrainingInput(train_uri, content_type=\"csv\")\n",
+    "test_uri = S3Uploader.upload(\"test_features.csv\", \"s3://{}/{}\".format(bucket, prefix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train XGBoost Model\n",
+    "#### Train Model\n",
+    "Since our focus is on understanding how to use SageMaker Clarify, we keep it simple by using a standard XGBoost model. For this section we will be using Amazon SageMaker Python SDK for simplicity."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "INFO:sagemaker:Creating training-job with name: sagemaker-xgboost-2023-01-17-21-41-16-380\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "2023-01-17 21:41:16 Starting - Starting the training job......\n",
+      "2023-01-17 21:41:49 Starting - Preparing the instances for training......\n",
+      "2023-01-17 21:42:26 Downloading - Downloading input data....\n",
+      "2023-01-17 21:42:51 Training - Downloading the training image....\n",
+      "2023-01-17 21:43:16 Training - Training image download completed. Training in progress.......\n",
+      "2023-01-17 21:43:51 Uploading - Uploading generated training model.\n",
+      "2023-01-17 21:44:02 Completed - Training job completed\n"
+     ]
+    }
+   ],
+   "source": [
+    "from sagemaker.image_uris import retrieve\n",
+    "from sagemaker.estimator import Estimator\n",
+    "\n",
+    "# This is references the AWS managed XGBoost container\n",
+    "xgboost_image_uri = retrieve(region=region, framework=\"xgboost\", version=\"1.5-1\")\n",
+    "\n",
+    "xgb = Estimator(\n",
+    "    xgboost_image_uri,\n",
+    "    role,\n",
+    "    instance_count=1,\n",
+    "    instance_type=\"ml.m5.xlarge\",\n",
+    "    disable_profiler=True,\n",
+    "    sagemaker_session=sagemaker_session,\n",
+    ")\n",
+    "\n",
+    "xgb.set_hyperparameters(\n",
+    "    max_depth=5,\n",
+    "    eta=0.2,\n",
+    "    gamma=4,\n",
+    "    min_child_weight=6,\n",
+    "    subsample=0.8,\n",
+    "    objective=\"binary:logistic\",\n",
+    "    num_round=800,\n",
+    ")\n",
+    "\n",
+    "xgb.fit({\"train\": train_input}, logs=False)"
```

Comment:
Good to warn the user how long the training job runs. E.g. "It takes about 10 minutes for the model to be trained."